### PR TITLE
Support for returning past_key_values from the model

### PR DESCRIPTION
**Pull request by idanshen**
The current implementation allows the passing of a KV cache into the forward functions of the Q\Value models but does not get the KV cache of such a call. This limits the ability to use the KV cache with these models.

### CODE REVIEW

File: trl/models/modeling_value_head.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ def forward(
         self,
         input_ids=None,
         past_key_values=None,
+        return_past_key_values=False,
```
Comment (younesbelkada) [inline trl/models/modeling_value_head.py:146]: Can you put that at the end (before kwargs)?

File: trl/models/modeling_value_head.py [hunk 2]
```diff
@@ -405,6 +410,7 @@ def forward(
...
```
Comment (younesbelkada) [inline trl/models/modeling_value_head.py:413]: same comment here

**Review by younesbelkada — COMMENTED** [review]
Thanks for extending the abilities of the value head modeling classes to output ! Just one comment about args ordering

File: trl/models/modeling_value_head.py (hunk 1 above)
Comment (idanshen) [reply trl/models/modeling_value_head.py]: Done

File: trl/models/modeling_value_head.py (hunk 2 above)
Comment (idanshen) [reply trl/models/modeling_value_head.py]: Done

**Comment by idanshen** [comment]
I change the order of variables, and synced the branch with main

**Review by younesbelkada — APPROVED** [review]
Thanks ! Failing tests are unrelated and will be fixed in https://github.com/huggingface/trl/pull/1750

**Pull request merged**